server/api.go

```diff
@@ -309,17 +314,17 @@ func (a *API) disconnect(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	userID := r.Header.Get("Mattermost-User-ID")
-	teamsUserID, err := a.p.store.MattermostToTeamsUserID(userID)
-	if err != nil {
-		a.p.API.LogError("The account is not connected.", "UserID", userID, "Error", err.Error())
+	userID := r.Header.Get(HeaderMattermostUserID)
+	if _, err := a.p.store.GetTokenForMattermostUser(userID); err != nil {
+		a.p.API.LogError("Unable to get the oauth token for the user.", "UserID", userID, "Error", err.Error())
 		http.Error(w, "The account is not connected.", http.StatusBadRequest)
```

Comment:
i think we can create a middleware for this logic as well, as it can be used in other APIs as well.